tests/integration/utils.py

```diff
@@ -93,6 +93,35 @@ def wait_for_complete_request(self, response):
 
         return response
 
+    def fetch_all_requests(self, request_json=None):
+        """
+        Fetch a list of requests from the Cachito API.
+
+        :param dict request_json: Request parameters and values (page, per_page, status, verbose)
+        :return: Object that contains response from the Cachito API
+        :rtype: list
+        """
+        if not request_json:
+            request_json = {}
+        result_url = f"{self._cachito_api_url}/requests"
+        resp = requests.get(result_url, json=request_json)
+        resp.raise_for_status()
+
+        page = request_json["page"]
+        if not page:
+            all_items = resp.json()["items"]
+            while True:
+                next_page = resp.json()["meta"]["next"]
+                if next_page is None:
+                    break
+
+                resp = requests.get(next_page)
+                resp.raise_for_status()
+                all_items += resp.json()["items"]
+            return Response({"items": all_items}, None, resp.status_code)
+
+        return Response(resp.json(), None, resp.status_code)
```

Comment:
How about this approach? It's a little clearer in my opinion and then the `Response` objects both have the the same format (i.e. just the `items` key set`).
```suggestion
        all_items = []
        while result_url:
            resp = requests.get(result_url, json=request_json, timeout=15)
            resp.raise_for_status()
            all_items += resp.json()["items"]
            result_url = resp.json()["meta"]["next"]

        return Response({"items": all_items}, None, resp.status_code)
```

Reply:
Wow, I really like it. Thank you!